tls/s2n_connection_serialize.c

```diff
@@ -219,6 +219,9 @@ static S2N_RESULT s2n_connection_deserialize_parse(uint8_t *buffer, uint32_t buf
     RESULT_GUARD_POSIX(s2n_stuffer_read_bytes(&input, protocol_version, S2N_TLS_PROTOCOL_VERSION_LEN));
     parsed_values->protocol_version = (protocol_version[0] * 10) + protocol_version[1];
 
+    RESULT_ENSURE(parsed_values->protocol_version >= S2N_SSLv3 && parsed_values->protocol_version <= S2N_TLS13,
+            S2N_ERR_PROTOCOL_VERSION_UNSUPPORTED);
```

Comment:
I wonder if using S2N_TLS13 as the upper bound is too brittle. Like, if we add a hypothetical TLS1.4, we'll probably be able to serialize it but we'll be unable to de-serialize it for no legitimate reason. That's why in most places in our code we check for >=TLS1.3 instead of ==TLS1.3.

Can we use [s2n_highest_protocol_version](https://github.com/aws/s2n-tls/blob/aac581bc1e007e41231defb4c4971a2952b4b9b6/tls/s2n_tls.h#L24C1-L24C45) instead?